code/modules/client/verbs/looc.dm

```diff
@@ -1,104 +1,86 @@
-
-        if (isobserver(M))
-            continue //Also handled later.
-
-        if(C.prefs.toggles & CHAT_OOC)
-//            var/display_name = src.key
-//            if(holder)
-//                if(holder.fakekey)
-//                    if(C.holder)
-//                        display_name = "[holder.fakekey]/([src.key])"
-//                else
-//                    display_name = holder.fakekey
-            to_chat(C,"<span class='looc'><span class='prefix'>LOOC:</span> <EM>[src.mob.name]:</EM> <span class='message'>[msg]</span></span>")
-
-    for(var/client/C in GLOB.admins)
-        if(C.prefs.toggles & CHAT_OOC)
-            var/prefix = "(R)LOOC"
-            if (C.mob in heard)
-                prefix = "LOOC"
-            to_chat(C,"<span class='looc'>[ADMIN_FLW(usr)]<span class='prefix'>[prefix]:</span> <EM>[src.key]/[src.mob.name]:</EM> <span class='message'>[msg]</span></span>")
+	set name = "LOOC"
+	set desc = "Local OOC, seen only by those in view."
+	set category = "OOC"
+
+	if(GLOB.say_disabled)    //This is here to try to identify lag problems
+		to_chat(usr, "<span class='danger'> Speech is currently admin-disabled.</span>")
+		return
+
+	if(!mob?.ckey)
+		return
+
+	msg = trim(sanitize(msg), MAX_MESSAGE_LEN)
+	if(!length(msg))
+		return
+
+	var/raw_msg = msg
+
+	if(!(prefs.toggles & CHAT_OOC))
+		to_chat(src, "<span class='danger'>You have OOC (and therefore LOOC) muted.</span>")
+		return
+
+	if(is_banned_from(mob.ckey, "OOC"))
+		to_chat(src, "<span class='danger'>You have been banned from OOC and LOOC.</span>")
+		return
+
+	if(!holder)
+		if(!CONFIG_GET(flag/looc_enabled))
+			to_chat(src, "<span class='danger'>LOOC is disabled.</span>")
+			return
+		if(!GLOB.dooc_allowed && (mob.stat == DEAD))
+			to_chat(usr, "<span class='danger'>LOOC for dead mobs has been turned off.</span>")
+			return
+		if(prefs.muted & MUTE_OOC)
+			to_chat(src, "<span class='danger'>You cannot use LOOC (muted).</span>")
+			return
+		if(handle_spam_prevention(msg, MUTE_OOC))
+			return
+		if(findtext(msg, "byond://"))
+			to_chat(src, "<B>Advertising other servers is not allowed.</B>")
+			log_admin("[key_name(src)] has attempted to advertise in LOOC: [msg]")
+			return
+		if(mob.stat)
+			to_chat(src, "<span class='danger'>You cannot salt in LOOC while unconscious or dead.</span>")
+			return
+		if(isdead(mob))
+			to_chat(src, "<span class='danger'>You cannot use LOOC while ghosting.</span>")
+			return
+		if(OOC_FILTER_CHECK(raw_msg))
+			to_chat(src, "<span class='warning'>That message contained a word prohibited in OOC chat! Consider reviewing the server rules.\n<span replaceRegex='show_filtered_ooc_chat'>\"[raw_msg]\"</span></span>")
+			return
+
+	msg = emoji_parse(msg)
+
+	mob.log_talk(raw_msg, LOG_OOC, tag="LOOC")
+
+	// Search everything in the view for anything that might be a mob, or contain a mob.
+	var/list/client/targets = list()
+	for(var/atom/movable/thingymajig in view(get_turf(mob)))
```

Comment:
Use appropriate variable names which give information about what is being targeted by this list

Reply:
Would it be quicker to loop through all clients and check their distance rather than recursively check all atoms in view?